components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/handlers/UsernameValidationConfigurationHandler.java

```diff
@@ -0,0 +1,162 @@
+
+    @Override
+    public boolean canHandle(String field) {
+
+        return USERNAME.equalsIgnoreCase(field);
+    }
+
+    @Override
+    public ValidationConfiguration getDefaultValidationConfiguration(String tenantDomain) throws
+            InputValidationMgtException {
+
+        ValidationConfiguration configuration = new ValidationConfiguration();
+        configuration.setField(USERNAME);
+        List<RulesConfiguration> rules = new ArrayList<>();
+
+        try {
+            RealmConfiguration realmConfiguration = getRealmConfiguration(tenantDomain);
+            Map<String, String> usernameRegEx = getUsernameRegEx(realmConfiguration);
+
+            // Return the JsRegex if the default regex has been updated by the user.
+            if (!usernameRegEx.get(JAVA_REG_EX).isEmpty() && !usernameRegEx.get(JS_REG_EX).isEmpty() &&
+                    !DEFAULT_EMAIL_REGEX_PATTERN.equals(usernameRegEx.get(JAVA_REG_EX))) {
+                rules.add(getRuleConfig("JsRegExValidator", JS_REGEX, usernameRegEx.get(JS_REG_EX)));
+                configuration.setRegEx(rules);
+            } else {
+                rules.add(getRuleConfig(EmailFormatValidator.class.getSimpleName(),
+                        ENABLE_VALIDATOR, Boolean.TRUE.toString()));
+                configuration.setRules(rules);
+            }
+            return configuration;
+        } catch (InputValidationMgtException e) {
+            throw new InputValidationMgtException(ERROR_GETTING_EXISTING_CONFIGURATIONS.getCode(), e.getMessage());
+        }
+    }
+
+    @Override
+    public boolean validateValidationConfiguration(List<RulesConfiguration> configurationList)
+            throws InputValidationMgtClientException {
+
+        List<String> validatorNames = new ArrayList<>();
+        configurationList.forEach(config -> validatorNames.add(config.getValidatorName()));
+        int validConfigurations = 0;
+        for (RulesConfiguration configuration: configurationList) {
+            if (EmailFormatValidator.class.getSimpleName().equals(configuration.getValidatorName())) {
+                if (Boolean.parseBoolean(configuration.getProperties().get(ENABLE_VALIDATOR))) {
+                    validConfigurations += 1;
+                }
+                validatorNames.remove(EmailFormatValidator.class.getSimpleName());
+            } else if (AlphanumericValidator.class.getSimpleName().equals(configuration.getValidatorName())) {
+                if (Boolean.parseBoolean(configuration.getProperties().get(ENABLE_VALIDATOR))) {
+                    if (validatorNames.contains(LengthValidator.class.getSimpleName())) {
+                        validConfigurations += 1;
+                        validatorNames.remove(LengthValidator.class.getSimpleName());
+                    }
+                }
+                validatorNames.remove(AlphanumericValidator.class.getSimpleName());
```

Comment:
Add a log mentioning that we can't handle the request as the length validator is not set along with the alphanumeric validator.

Reply:
Addressed